Can't build for aarch64-linux-android

```
cd openat
$ cargo build --target=aarch64-linux-android
    Updating crates.io index
 Downloading libc v0.2.45
   Compiling libc v0.2.45
   Compiling openat v0.1.16 (/data/data/com.termux/files/home/openat)
error[E0425]: cannot find function __error in module libc
  --> src/list.rs:47:11
   |
47 |     libc::__error()
   |           ^^^^^^^ did you mean __errno?

error: aborting due to previous error
```

--- a/src/list.rs
+++ b/src/list.rs
@@ -37,12 +37,12 @@ unsafe fn errno_location() -> *mut libc::c_int {
     libc::__errno_location()
 }
 
-#[cfg(any(target_os="openbsd", target_os="netbsd"))]
+#[cfg(any(target_os="openbsd", target_os="netbsd", target_os="android"))]
 unsafe fn errno_location() -> *mut libc::c_int {
     libc::__errno()
 }
 
-#[cfg(not(any(target_os="linux", target_os="openbsd", target_os="netbsd")))]
+#[cfg(not(any(target_os="linux", target_os="openbsd", target_os="netbsd", target_os="android")))]
 unsafe fn errno_location() -> *mut libc::c_int {
     libc::__error()
 }

